dss/events/handlers/index.py

```diff
@@ -218,9 +221,21 @@ def notify(subscription_id: str, subscription: dict, bundle_id: str, logger):
             "bundle_version": bundle_version
         }
     }
-    # TODO (mbaumann) Ensure webhooks are only delivered over verified HTTPS (unless maybe when running a test)
     callback_url = subscription['callback_url']
-    response = requests.post(callback_url, json=payload)
+
+    assert urlparse(callback_url).scheme in {"http", "https"}
+    if os.environ["DSS_DEPLOYMENT_STAGE"] not in {"dev", "staging"}:
+        hostname = urlparse(callback_url).hostname
+        for family, socktype, proto, canonname, sockaddr in socket.getaddrinfo(hostname, port=None):
+            assert ipaddress.ip_address(sockaddr[0]).is_global  # type: ignore
+        assert urlparse(callback_url).scheme == "https"
+
+    auth = None
+    if "hmac_secret_key" in subscription:
+        auth = HTTPSignatureAuth(key=subscription["hmac_secret_key"].encode(),
```

Comment:
My understanding from ttung is our convention is to use single quotes for dict key names, and double quotes for all other strings.

Reply:
In Python, there is no convention about which quoting styles to use where. It's important to be consistent though, so I've changed this to single quotes.